nexus/src/app/sagas/instance_create.rs

```diff
@@ -1310,58 +1365,72 @@ async fn sic_instance_ensure(
     } else {
         osagactx
             .nexus()
-            .instance_set_runtime(
-                &opctx,
-                &authz_instance,
-                &db_instance,
-                InstanceStateRequested::Running,
-            )
+            .instance_ensure_registered(&opctx, &authz_instance, &db_instance)
             .await
             .map_err(ActionError::action_failed)?;
     }
 
     Ok(())
 }
 
-async fn sic_noop(_sagactx: NexusActionContext) -> Result<(), ActionError> {
-    Ok(())
-}
-
-/// Ensure that the necessary v2p mappings exist for this instance
-async fn sic_v2p_ensure(
+async fn sic_instance_ensure_registered_undo(
     sagactx: NexusActionContext,
-) -> Result<(), ActionError> {
+) -> Result<(), anyhow::Error> {
```

Comment:
Our hand's forced here because Steno (currently) defines undo actions to return `anyhow::Error`: https://docs.rs/steno/latest/steno/type.UndoResult.html